atlp-rwanda/phantom-fe-codebandits
# ch-(tailwind)-Integrate tailwind-cp8

## What this PR do:

- [x] Successfully add tailwind CSS to the project
- [x] Replace all prior styles with the tailwind classes
- [x] Successfully add primary colors  to the theme

## Description of Task to be completed
Set up Tailwind CSS in the React app

- [x] Tailwind CSS is setup in the project
- [x] Sample styles are rendering using Tailwind CSS

### How should this be manually tested?
Check the prior requirements on our [page here](https://github.com/atlp-rwanda/phantom-fe-codebandits#readme)
```
// build UI components and use the tailwind CSS
// Linked below are the resources
npm i
npm start
// check your renders for applied styles
```
**[Tailwind states](https://tailwindcss.com/docs/hover-focus-and-other-states)**
## Recommendations

For better development workflow, I recommend:
- Installing [Tailwind Extension for VS Code from here](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss)
- Change VS Code settings to allow `@ rules for CSS` by adding this entry into `settings.json` < `"css.lint.unknownAtRules": "ignore",`.> **To access settings**: press `CTRL + P` then type `settings.json`

## Any background context you want to provide?
The integration was successfully with the use of **Postcss** as the web-pack loader
According to the docs, Tailwind require no **CSS preprocessor** but given the nature of the project the use was required.

As an addition:
**The following files has to be configure as `common js` files to avoid web-pack errors. It does not affect their role.** 
- **tailwing.config.js** to **`tailwind.config.cjs`**
- **postcss.config.js** to **`postcss.config.cjs`**

> Trying to convert these files into ES6 modules broke the functionality


## What are the relevant stories
- **[#cp8](https://trello.com/c/LKBFgNmn)**
